libslide/src/math/gcd_poly_zz.rs

```diff
@@ -0,0 +1,430 @@
+use crate::math::gcd;
+
+use core::cmp::{max, min};
+
+#[allow(unused)]
+type Poly = Vec<isize>;
+
+#[allow(unused)]
+type PolyPtrArg = [isize];
```

Comment:
Instead of this just use `&Poly`

Reply:
Clippy yelled at me to do this

Reply:
damn she dummy, clippy don't you understand nominal types? Anyway I stand by the struct idea which would make this obsolete anyway